### PR TITLE
findOneBy added to Repository

### PR DESCRIPTION
**Pull request by asev**
This is useful method for repository.

### CODE REVIEW

File: ORM/Repository.php [hunk 1]
```diff
@@ -160,6 +160,32 @@ public function findBy(
...
 
     /**
+     * Finds only one entity by a set of criteria.
+     *
+     * @param array      $criteria Example: ['group' => ['best', 'worst'], 'job' => 'medic'].
+     * @param array|null $orderBy  Example: ['name' => 'ASC', 'surname' => 'DESC'].
+     *
+     * @return DocumentInterface|null The object.
+     */
+    public function findOneBy(array $criteria, array $orderBy = [])
```
Comment (saimaz) [inline ORM/Repository.php]: IMO there should be a third argument with result type.

**Comment by martiis** [comment]
Issue #346

**Comment by tautelis** [comment]
Looking forward for the release with this useful method!

**Comment by asev** [comment]
I've added resultType argument to findOneBy(). In case when resultType is RESULTS_OBJECT - it acts the same as find().

**Comment by asev** [comment]
This one is still important for our project. Do you see any problems with implementation?

**Comment by saimaz** [comment]
Look ok.

**Pull request merged**